lib/platform/bitbucket/index.ts

```diff
@@ -116,7 +116,8 @@ export async function getRawFile(
   // See: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/src/%7Bcommit%7D/%7Bpath%7D
   const repo = repoName ?? config.repository;
   const path = fileName;
-  const url = `/2.0/repositories/${repo}/src/HEAD/${path}`;
+  const url =
+    `/2.0/repositories/${repo}/src/` + (branchOrTag || `HEAD`) + `/${path}`;
```

Comment:
I think this needs some real tests, as i'm not sure if branches needs to be encoded, or if we need to fetch the commit sha first

Reply:
Hi @viceice I've done real testing on my repo https://bitbucket.org/olegkrivtsov/renovate-test. 

I used these API calls for getting the file contents:

curl --request GET --url 'https://olegkrivtsov:XXXXXXXX@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/dev/README.md'

curl --request GET --url 'https://olegkrivtsov:m3DgrS7DYRBKgMVQX9yG@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/00fa6d4b0ebcb3bcd9940a0cfe436185a2e32b66/README.md'

Both of them worked, so it seems we can pass both branch name (no encoding is needed), and SHA1 of a commit.

Reply:
Can you test a branch with slashes?

Reply:
Hi @viceice I've tested a branch with slash `feat/123-test`:

```
curl --request GET --url 'https://olegkrivtsov:XXXXX@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/feat/123-test/README.md'
```

This didn't work.

I tried to url-encode the branch name, and it didn't work as well:

```
curl --request GET --url 'https://olegkrivtsov:gaM5YcN8nw9fyd7Esvvd@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/feat%2F123-test/README.md'
```

I guess that maybe getting the SHA1 of a HEAD commit of the branch and passing it to the URL would work?

Reply:
ok, so branches whithout slash can used directly and others needs to fetch head commit first. Can you implement this?

Reply:
@viceice sure, I'll implement this.